### PR TITLE
Fix #18: Scala.JS support

### LINKED ISSUE #18: Scala.JS support
I would like to check how much interest there is in taking Scala.JS seriously in cats.

That might not be an easy decision to make but it could actually be :-)

### PR DESCRIPTION
**Pull request by InTheNow**
Adds scalajs support

### CODE REVIEW

**Comment by non** [comment]
Awesome! @InTheNow do you know what the deal with the build failure is?

**Comment by InTheNow** [comment]
Build failure is due to "Node.js isn't connected" see https://github.com/scala-js/scala-js/issues/1546 . I've pushed a new build, if that fails, I'll try a few more tricks

**Comment by non** [comment]
It's green! :birthday: 

I'm :+1: to merge this

File: laws/js/src/main/scala/cats/laws/Platform.scala [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package cats
+package laws
+
+import org.scalacheck.{Arbitrary, Prop}
+import org.scalacheck.Prop._
+import Prop.{False, Proof, Result}
+
+private[laws] object Platform {
+
+  // Scala-js does not implement the Serializable interface, so we just retuen true.
```
Comment (martin-g) [inline laws/js/src/main/scala/cats/laws/Platform.scala]: s/retuen/return/

Comment (InTheNow) [inline laws/js/src/main/scala/cats/laws/Platform.scala]: Can fix now, but if this is the only issue, I would rather add to https://github.com/non/cats/issues/429.

Well spotted, though!

File: tests/js/src/test/scala/cats/tests/Platform.scala [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package cats
+package tests
+
+private[tests] object Platform {
+
+  trait UltraSlowCatsSuite extends CatsSuite {
+    implicit override val generatorDrivenConfig: PropertyCheckConfiguration =
+    PropertyCheckConfig(maxSize = 1, minSuccessful = 1)
```
Comment (ceedubs) [inline tests/js/src/test/scala/cats/tests/Platform.scala]: This is quite dodgy :)

Comment (ceedubs) [inline tests/js/src/test/scala/cats/tests/Platform.scala]: Ah I see, we are only doing this for JS, not JVM. :+1:

**Comment by ceedubs** [comment]
:tada: :+1:

**Comment by ceedubs** [comment]
Thanks a bunch, @InTheNow!

**Comment by ngbinh** [comment]
can't wait for a new release with this.

**Pull request merged**